core/liblwm2m.c

```diff
@@ -441,11 +443,14 @@ int lwm2m_step(lwm2m_context_t * contextP,
             break;
 
         case STATE_REG_FAILED:
+            contextP->remaining_dataLen = 0;
+            contextP->block_no = 0;
+        case STATE_REG_PARTIAL_DONE:
```

Comment:
Why would this cause failure? Seems to me it should send the next block. Maybe I'm just missing where the next block gets sent.

Reply:
There are 2 cases.
1. When there is a failure in block transfer of registration string and we need to reset the block_no and remaining_dataLen.
2. When there is blocks have been partially transferred.

In either case, registration_start needs to be called and based on block_no and remaining_dataLen, the respective block data gets transferred.

Reply:
It looks like case 2 would start a bootstrap rather than sending the next block. I think it should go to STATE_REGISTER_REQUIRED to call registration_start rather than STATE_BOOTSTRAP_REQUIRED.

Reply:
Point noted, will change this